### PR TITLE
Migrate the event stream iteration logic to ballerina side.

### PR DESCRIPTION
**Pull request by MohamedSabthar**
## Purpose
$ Subject
This is an improvement done to the previous PR: https://github.com/ballerina-platform/module-ballerina-mime/pull/528
## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: ballerina/event_stream_writer.bal [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    isolated function init(stream<byte[], error?> eventStream) {
+        self.eventStream = eventStream;
+    }
+
+    isolated function writeEventStream() returns error? {
+        var iterator = self.eventStream.iterator();
+        do {
+            while true {
+                record {byte[] value;}? event = check iterator.next();
```
Comment (TharmiganK) [inline ballerina/event_stream_writer.bal:31]: Do we need the iterator here? How about this:
```suggestion
                record {byte[] value;}? event = check self.eventStream.next();
```

File: ballerina/event_stream_writer.bal [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                if event is () {
+                    return;
+                }
+                check trap self.writeEventStreamBytesToOutputStream(event.value);
```
Comment (TharmiganK) [inline ballerina/event_stream_writer.bal:35]: Can't we directly call the external function?
```suggestion
                check trap externWriteEventStreamBytesToOutputStream(self, event.value);
```

File: ballerina/event_stream_writer.bal [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import ballerina/log;
+
+class EventStreamWriter {
+    stream<byte[], error?> eventStream;
+
+    isolated function init(stream<byte[], error?> eventStream) {
+        self.eventStream = eventStream;
+    }
+
+    isolated function writeEventStream() returns error? {
```
Comment (TharmiganK) [inline ballerina/event_stream_writer.bal:27]: This function does not return an error right?
```suggestion
    isolated function writeEventStream() {
```

File: ballerina/event_stream_writer.bal (hunk 2 above)
Comment (TharmiganK) [inline ballerina/event_stream_writer.bal:35]: In this write function, where are we panic an error?

File: ballerina/event_stream_writer.bal [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        var iterator = self.eventStream.iterator();
+        do {
+            while true {
+                record {byte[] value;}? event = check iterator.next();
+                if event is () {
+                    return;
```
[the remark this answers is not in the document]
Comment (MohamedSabthar) [reply ballerina/event_stream_writer.bal:33]: yes, we need to close. Closed.

**Review by TharmiganK — APPROVED** [review]

**Pull request merged**